File: Length_of_Last_Word/Solution2.java
/* 
	Solution 2 (LEETCODE SOLUTION)
	
	Simply use Trim() to get rid of leading/trailing whitespaces, then Split(" ") to separate all
	sequences divided by whitespace, then return the final string in the split array's length.
*/


/* 
	Time Complexity: O() 
	Space Complexity: O()
*/

public class Solution {
    public int LengthOfLastWord(string s) {
        string[] splitString = s.Trim(' ').Split(" ");
        return splitString[splitString.Length - 1].Length;
    }
}

/*
	LEET CODE RESULTS

	Runtime: 109 ms, faster than 15.54% of C# online submissions for Length of Last Word.
	Memory Usage: 22.8 MB, less than 48.52% of C# online submissions for Length of Last Word.
*/

File: Missing_Number/Solution2.cs
/* 
	Solution 2 
	
	The sum of numbers 1 through n should equal to (n * (n+1)) / 2 .
	Simply sum all of the entries and return the expected result - the actual result. The missing
	number should be the difference.
*/


/* 
	Time Complexity: O(n)
	Space Complexity: O(1)
*/

public class Solution {
    public int MissingNumber(int[] nums) {
        double sum = 0;
        foreach(int num in nums) sum+=num;
        int len = nums.Length;
        double expected = (len * (len+ 1)) / 2;
        return (int) (expected - sum);
    }
}

/*
	LEET CODE RESULTS

	Runtime: 104 ms, faster than 91.74% of C# online submissions for Missing Number.
	Memory Usage: 30.5 MB, less than 22.19% of C# online submissions for Missing Number.
*/